code/game/jobs/job/marine/squads.dm

```diff
@@ -55,43 +83,136 @@
 	var/obj/item/device/squad_beacon/bomb/bbeacon = null
 	var/obj/structure/supply_drop/drop_pad = null
 
-/datum/squad/alpha
-	name = SQUAD_NAME_1
+
+/datum/squad/marine
+	usable = TRUE
+	active = TRUE
+	faction = FACTION_MARINE
+	lead_icon = "leader"
+
+/datum/squad/marine/alpha
+	name = SQUAD_MARINE_1
 	color = 1
 	access = list(ACCESS_MARINE_ALPHA)
-	usable = TRUE
 	radio_freq = ALPHA_FREQ
 
-/datum/squad/bravo
-	name = SQUAD_NAME_2
+/datum/squad/marine/bravo
+	name = SQUAD_MARINE_2
 	color = 2
 	access = list(ACCESS_MARINE_BRAVO)
-	usable = TRUE
 	radio_freq = BRAVO_FREQ
 
-/datum/squad/charlie
-	name = SQUAD_NAME_3
+/datum/squad/marine/charlie
+	name = SQUAD_MARINE_3
 	color = 3
 	access = list(ACCESS_MARINE_CHARLIE)
-	usable = TRUE
 	radio_freq = CHARLIE_FREQ
 
-/datum/squad/delta
-	name = SQUAD_NAME_4
+/datum/squad/marine/delta
+	name = SQUAD_MARINE_4
 	color = 4
 	access = list(ACCESS_MARINE_DELTA)
-	usable = TRUE
 	radio_freq = DELTA_FREQ
 
-/datum/squad/echo
-	name = SQUAD_NAME_5
+/datum/squad/marine/echo
+	name = SQUAD_MARINE_5
 	color = 5
 	access = list(ACCESS_MARINE_ALPHA, ACCESS_MARINE_BRAVO, ACCESS_MARINE_CHARLIE, ACCESS_MARINE_DELTA)
+	radio_freq = ECHO_FREQ
+	omni_squad_vendor = TRUE
+
+	active = FALSE
+	roundstart = FALSE
+	locked = TRUE
+
+/datum/squad/marine/cryo
+	name = SQUAD_MARINE_CRYO
+	color = 6
+	access = list(ACCESS_MARINE_ALPHA, ACCESS_MARINE_BRAVO, ACCESS_MARINE_CHARLIE, ACCESS_MARINE_DELTA)
 	usable = TRUE
 	roundstart = FALSE
 	omni_squad_vendor = TRUE
-	radio_freq = ECHO_FREQ
+	radio_freq = CRYO_FREQ
+	locked = TRUE
+
+/datum/squad/marine/marsoc
+	name = SQUAD_MARSOC
+	color = 7
+	usable = TRUE
+	roundstart = FALSE
+	radio_freq = MARSOC_FREQ
+	locked = TRUE
+	squad_type = "Team"
+	lead_icon = "soctl"
+
+//############################### UPP Squads
+/datum/squad/upp
+	name = "UPP Root Squad"
+	usable = TRUE
+	omni_squad_vendor = TRUE
+	faction = FACTION_UPP
+
+/datum/squad/upp/one
+	name = "UPPS1"
+	color = 1
+
+/datum/squad/upp/twp
+	name = "UPPS2"
+	color = 2
+
+/datum/squad/upp/three
+	name = "UPPS3"
+	color = 3
 
+/datum/squad/upp/four
+	name = "UPPS4"
+	color = 4
+
+/datum/squad/upp/kdo
+	name = "UPPKdo"
+	color = 6
+	squad_type = "Team"
+	locked = TRUE
+//###############################
```

Comment:
the whitespaces between the lines here are annoying, just put a single whitespace between every entry instead of sometimes 1, sometimes 0